pkg/apis/cluster/v1alpha1/conversions/conversions.go

```diff
@@ -0,0 +1,73 @@
+package conversions
+
+import (
+	"encoding/json"
+	"fmt"
+
+	machinesv1alpha1 "github.com/kubermatic/machine-controller/pkg/machines/v1alpha1"
+
+	clusterv1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+)
+
+func Convert_MachinesV1alpha1Machine_To_ClusterV1alpha1Machine(in *machinesv1alpha1.Machine, out *clusterv1alpha1.Machine) error {
+	out.ObjectMeta = in.ObjectMeta
+	out.Spec.ObjectMeta = in.Spec.ObjectMeta
+	out.SelfLink = ""
+	out.UID = ""
+	out.ResourceVersion = ""
+	out.Generation = 0
+	out.CreationTimestamp = metav1.Time{}
```

Comment:
Resetting the above fields is required as those must only be populated by the apiserver?

Reply:
Yes, they are all read-only: https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#ObjectMeta